OpenRA.Mods.Common/Traits/AppearsOnRadar.cs

```diff
@@ -40,7 +40,22 @@ public void Created(Actor self)
 
 		public IEnumerable<Pair<CPos, Color>> RadarSignatureCells(Actor self)
 		{
-			var color = modifier != null ? modifier.RadarColorOverride(self) : self.Owner.Color.RGB;
+			var player = self.World.RenderPlayer ?? self.World.LocalPlayer;
+			var color = self.Owner.Color.RGB;
+
+			if (Game.Settings.Game.AllianceDisplay)
+				if (self.Owner.PlayerName != "Neutral")
+				{
```

Comment:
Style nit: avoid hanging blocks.  If you can't combine these nested conditionals, you should add braces around the outer block too.